### PR TITLE
Support `deprecated` attribute since Vulkan 1.3.241

### PR DESCRIPTION
**Pull request by MarijnS95**

### CODE REVIEW

File: vk-parse/src/parse.rs [hunk 1]
```diff
@@ -480,7 +481,8 @@ fn parse_type<R: Read>(ctx: &mut ParseCtx<R>, attributes: Vec<XmlAttribute>) ->
         "allowduplicate" => allowduplicate = Some(a.value),
         "objtypeenum"    => objtypeenum    = Some(a.value),
         "bitvalues"      => bitvalues      = Some(a.value),
-        "comment"        => comment        = Some(a.value)
+        "comment"        => comment        = Some(a.value),
+        "deprecated"     => deprecated     = a.value.parse().ok(),
```
Comment (MarijnS95) [inline vk-parse/src/parse.rs:485]: TODO: Need to append this exception to `$ctx` before setting `deprecated` to `None`; but it doesn't seem `FromStr` with `.parse()` has been used before.

File: vk-parse/src/types.rs [hunk 2]
```diff
@@ -106,6 +127,24 @@ pub enum RegistryChild {
...
+    ///
+    /// Denotes that the enum or type likely contains a typo and only exists to maintain backwards compatibility, aliasing a renamed base type or enum variant.
+    Aliased,
+    /// Valid on enums and members
+    ///
+    /// Denotes that this member or enum variant is ignored and should not be used.
+    Ignored,
+    /// Valid on enums, types and members
+    ///
+    /// This item is deprecated without specific explanation.
```
[the remark this answers is not in the document]
Comment (MarijnS95) [reply vk-parse/src/types.rs]: Found it in `registry.adoc`: https://github.com/KhronosGroup/Vulkan-Docs/commit/1b1c4dd43a35341c8c8e82ad985ed66d8beff5ba#diff-9d5e8331011b604632c0000219359014bd7abf8e883dca9827eb333e921b96b7

**Comment by krolli** [comment]
Is strongly typing values of `deprecated` attribute using enum necessary? I have moved away from doing that almost right away and it currently doesn't fit into the rest of API. I would prefer if it were just a string as everything else.

**Comment by MarijnS95** [comment]
@krolli I think that is fine, and explains why I did not see any value parsing before. That way `vk-parse` isn't nearly as tied to the XML (and documentation) schema as well, I'll drop it!

**Comment by krolli** [comment]
Ugh, I think I just implemented all three of your PRs in single commit.

TBH, these small things I usually get implemented reasonably soon. Sometimes it is more work for me to deal with PRs (review, respond, merge, test) than it is to add those attributes myself, though I know people mean it well.

**Comment by MarijnS95** [comment]
Ugh. You already knew this PR was open and "accidentally implemented it yourself". Sure.

**Comment by krolli** [comment]
I didn't know the other two were there. Was planning to rebase my commit onto yours, but wanted to get all errors out of the way so I know it's everything.

**Comment by krolli** [comment]
To be honest, I am becoming the biggest roadblock when it comes to adding support for new stuff in vulkan XML. While I know about new release adding unrecognized attributes or elements within a day, it may take several days for me to get around to implementing that stuff, which is usually just a few minutes of work. By that time, someone who needs the thing to be in (lately it seems to be you) already created a PR adding it. Not gonna lie, for these tiny changes, I don't really like dealing with PRs.

@MarijnS95, I am thinking of giving you better access to the repository, if you want it. I am well aware of myself being a bottleneck and I also lack the perspective of user of this library. I would still be making sure that new releases of XML spec parse fine (within a few days of the release), but if you or someone else add it in earlier, I don't want to be in the way of it getting into the repo. I might still say something before I publish new version of the crate though. Are you interested?

**Comment by MarijnS95** [comment]
@krolli understood. As we are Khronos members and have Ash codegen in the upstream CI I'm usually _ahead_ of the curve, actively collaborating with other members to make sure `vk.xml` is compatible with `ash`  (and `vk-parse`) _up front_. Not that I usually make changes to both repos up-front and PR them as soon as the new spec is public, though.

That said I'm a similar bottleneck. Ash works well for us and as long as we don't need anything specific I refrain from big changes beyond regular spec updates. And that's hampering recent efforts to merge with Erupt, which has superior tooling that we have yet to integrate with Ash (and may shy away from `vk-parse`, but discussions stagnated).

While I wouldn't mind having better access to this repo I also never blindly apply my own changes and always go through PRs that I expect peer review on, even for trivial submissions (and the trivial things typically break something subtle and unintended).

**Pull request closed without merging**